File: LeetCode/LeetCode.Medium/EvaluateReversePolishNotation.cs
namespace LeetCode.Medium
{
    public class EvaluateReversePolishNotation
    {
        public static int EvalRPN(string[] tokens)
        {
            var stack = new Stack<int>();
            foreach (var c in tokens)
            {
                if (c == "+")
                {
                    stack.Push(stack.Pop() + stack.Pop());
                }
                else if (c == "-")
                {
                    int a = stack.Pop();
                    int b = stack.Pop();
                    stack.Push(b - a);
                }
                else if (c == "*")
                {
                    stack.Push(stack.Pop() * stack.Pop());
                }
                else if (c == "/")
                {
                    int a = stack.Pop();
                    int b = stack.Pop();
                    stack.Push(b / a);
                }
                else
                {
                    stack.Push(int.Parse(c));
                }
            }
            return stack.Pop();
        }
    }
}

File: js/EvaluateReversePolishNotation.js
var evalRPN = function (tokens) {
  const stack = [];
  for (const c of tokens) {
    if (c === "+") {
      stack.push(stack.pop() + stack.pop());
    } else if (c === "-") {
      const a = stack.pop();
      const b = stack.pop();
      stack.push(b - a);
    } else if (c === "*") {
      stack.push(stack.pop() * stack.pop());
    } else if (c === "/") {
      const a = stack.pop();
      const b = stack.pop();
      stack.push(b / a);
    } else {
      stack.push(parseInt(c));
    }
  }

  return stack.pop();
};

tokens = ["1", "2", "+", "3", "*", "4", "-"];
console.log(evalRPN(tokens));